lib/rex/post/meterpreter/extensions/stdapi/net/resolve.rb

```diff
@@ -56,40 +55,31 @@ def resolve_hosts(hostnames, family=AF_INET)
 
     hosts = []
     raws = []
-    types = []
 
     response.each(TLV_TYPE_IP) do |raw|
       raws << raw
     end
 
-    response.each(TLV_TYPE_ADDR_TYPE) do |type|
-      types << type
-    end
-
     0.upto(hostnames.length - 1) do |i|
       raw = raws[i]
-      type = types[i]
       host = hostnames[i]
 
-      hosts << raw_to_host_ip_pair(host, raw.value, type.value)
+      hosts << raw_to_host_ip_pair(host, raw.value)
```

Comment:
```suggestion
      hosts << raw_to_host_ip_pair(host, raw&.value)
```

Save navigation as `nil` doesn't have a `value` method and `raw_to_host_ip_pair` will handle `raw` having a `nil` value appropriately.